### PR TITLE
[transform.math] Fix missing unit when result is 0

### LINKED ISSUE #506: [Math transform] Unit is ignored if result is 0
At night I observe this warning : 

`2023-07-21 22:24:54.597 [WARN ] [penhab.core.library.items.NumberItem] - Failed to update item 'Total_Radiation_COR' because '0' could not be converted to the item unit 'W/m²'`

The item definition is : 

```
Number:Intensity Total_Radiation_COR "Radiation pondérée" <radiation> (gAstroSun1,gSynopanalyzer1) ["Point"] {unit="W/m²", channel="astro:sun:1:radiation#total" [profile="transform:MULTIPLY", multiplicand=0, itemName="Synop_Attenuation"]}
```

It takes current solar radiation : 

```
Number:Intensity Total_Radiation "Radiation Totale" <radiation> (gAstroSun1) ["Point"] {unit="W/m²", channel="astro:sun:1:radiation#total"}
```
and multiplies it by the cloudiness factor (that has no unit) : 

```
Number Synop_Attenuation "Coefficient Atténuation" <sun_clouds> (gSynopanalyzer1) ["Measurement"] {channel="synopanalyzer:synopanalyzer:1:attenuation-factor"}
```

During the day, it works fine but at night Total_Radiation = 0 W/m² and the warning is emitted.
Taking a look at `AbstractMathTransformationService`, I found the origin of the problem : 

```
QuantityType<?> result = performCalculation(source, value);
return BigDecimal.ZERO.compareTo(result.toBigDecimal()) == 0 ? "0" : result.toString();
```
So, if the result is 0, the code drops the unit, leading to the warning.

I think this could be changed to : 

`
return performCalculation(source, value).toString();
`

@J-N-K : what do you think ?

### PR DESCRIPTION
**Pull request by clinique**
Resolves #506

### CODE REVIEW

File: bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/transform/math/internal/AbstractMathTransformationService.java [hunk 1]
```diff
@@ -41,22 +39,19 @@ abstract class AbstractMathTransformationService implements TransformationServic
...
         try {
-            value = new QuantityType<>(valueString);
+            source = new QuantityType<>(evaluated);
+            evaluated = valueString;
+            value = new QuantityType<>(evaluated);
         } catch (IllegalArgumentException e) {
-            logger.warn("Input value '{}' could not be converted to a valid number", valueString);
+            logger.warn("Input value '{}' could not be converted to a valid number", evaluated);
             throw new TransformationException("Math Transformation can only be used with numeric inputs");
         }
```
Comment (J-N-K) [inline bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/transform/math/internal/AbstractMathTransformationService.java]: Why did you change that?

File: bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/transform/math/internal/AbstractMathTransformationService.java [hunk 2]
```diff
@@ -41,22 +39,19 @@ abstract class AbstractMathTransformationService implements TransformationServic
...
+        
         try {
-            QuantityType<?> result = performCalculation(source, value);
-            return BigDecimal.ZERO.compareTo(result.toBigDecimal()) == 0 ? "0" : result.toString();
+            return performCalculation(source, value).toString();
```
Comment (J-N-K) [inline bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/transform/math/internal/AbstractMathTransformationService.java]: I See why it works now, but I wonder why it was done like that before. Maybe to prevent a return value of `0.0000`?

File: bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/transform/math/internal/AbstractMathTransformationService.java (hunk 1 above)
Comment (clinique) [reply bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/transform/math/internal/AbstractMathTransformationService.java]: To avoid having two catch catching the same thing....could be avoided.

File: bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/transform/math/internal/AbstractMathTransformationService.java (hunk 2 above)
Comment (clinique) [reply bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/transform/math/internal/AbstractMathTransformationService.java]: Yes, I was wondering the same, reason why I wanted your advice.

File: bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/transform/math/internal/AbstractMathTransformationService.java (hunk 1 above)
Comment (clinique) [reply bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/transform/math/internal/AbstractMathTransformationService.java]: Reverted

**Review by J-N-K — DISMISSED** [review]

**Review by J-N-K — APPROVED** [review]
